### PR TITLE
Update configuration deserialization to be fault-tolerant of missing env vars

### PR DESCRIPTION
**Pull request by nepoche**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
Added unused config with undefined environment variable in tests directory

- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Users define the networks they wish to support with the appropriate WEBB_EVM_network_ENABLED and network_PRIVATE_KEY environment variables. However, errors still appear for networks they do not intend to support, because the environment variables for private keys for these unsupported networks do not exist.

## What is the new behavior?
Incrementally deserialize each config file, and include the config to the relayer runtime if appropriately parsed. If deserialization fails for whatever reason, skip that config and continue.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,7 @@ thiserror = "^1.0"
 glob = "^0.3"
 headers = "0.3.5"
 dotenv = "0.15.0"
+regex = "1.5.4"
```
Comment (shekohex) [inline Cargo.toml:41]: Do we need `regex` now?

File: src/config.rs [hunk 2]
```diff
@@ -252,9 +252,25 @@ pub fn load<P: AsRef<Path>>(path: P) -> anyhow::Result<WebbRelayerConfig> {
     // then get an iterator over all matching files
     let config_files = glob::glob(&pattern)?.flatten();
     for config_file in config_files {
+        let base = config_file.display().to_string();
         let file = config::File::from(config_file).format(FileFormat::Toml);
-        // merge the file into the config
-        cfg.merge(file)?;
+        let mut incremental_config = config::Config::new();
+        incremental_config.merge(file.clone())?;
+        match incremental_config.try_into::<WebbRelayerConfig>() {
```
Comment (shekohex) [inline src/config.rs:259]: Could you use `serde_path_to_error::deserialize::<WebbRelayerConfig>(incremental_config)` instead? so we can have a path to the error too here.

File: src/config.rs [hunk 3]
```diff
@@ -252,9 +252,25 @@ pub fn load<P: AsRef<Path>>(path: P) -> anyhow::Result<WebbRelayerConfig> {
...
+                cfg.merge(file)?
+            }
+            Err(e) => {
+                // print the issue that occurred while deserializing, then skip that config
+                tracing::debug!(
+                    "Error {} while attempting to deserialize {}",
+                    e,
+                    base
+                );
+                &cfg
```
Comment (shekohex) [inline src/config.rs:271]: why do you return the `cfg` here?

**Review by shekohex — CHANGES_REQUESTED** [review]
Looks good, just left a few changes.

File: Cargo.toml (hunk 1 above)
Comment (nepoche) [reply Cargo.toml]: Removed it

File: src/config.rs (hunk 2 above)
Comment (nepoche) [reply src/config.rs]: I'm not sure why it required the types to be specified but the code did not have an issue with the later match against config. I updated it with types explicitly stated.

File: src/config.rs (hunk 3 above)
Comment (nepoche) [reply src/config.rs]: removed

**Pull request merged**